src/libaktualizr/primary/download_nonostree_test.cc

```diff
@@ -0,0 +1,89 @@
+#include <gtest/gtest.h>
+
+#include <future>
+#include <iostream>
+#include <string>
+#include <thread>
+
+#include <boost/process.hpp>
+
+#include "uptane_test_common.h"
+
+#include "config/config.h"
+#include "logging/logging.h"
+#include "package_manager/ostreemanager.h"
+#include "primary/aktualizr.h"
+#include "storage/sqlstorage.h"
+#include "test_utils.h"
+
+#include <ostree.h>
+
+boost::filesystem::path aktualizr_repo_path;
+static std::string server = "http://127.0.0.1:";
+static std::string treehub_server = "http://127.0.0.1:";
+static boost::filesystem::path sysroot;
+
+static std::string new_rev;
+
+/*
+ * Reject non-OSTree binaries.
+ */
+TEST(Aktualizr, InstallNonOstreeBin) {
+  TemporaryDirectory temp_dir;
+  Config conf = UptaneTestCommon::makeTestConfig(temp_dir, server);
+  conf.pacman.type = PackageManager::kOstree;
+  conf.pacman.sysroot = sysroot.string();
+  conf.pacman.ostree_server = treehub_server;
+  conf.pacman.os = "dummy-os";
+  conf.provision.device_id = "device_id";
+  conf.provision.ecu_registration_endpoint = server + "/director/ecus";
+  conf.tls.server = server;
+
+  {
+    std::shared_ptr<INvStorage> storage = INvStorage::newStorage(conf.storage);
+    OstreeManager ostree(conf.pacman, storage, nullptr, nullptr);
+
+    Aktualizr aktualizr(conf);
+
+    aktualizr.Initialize();
+
+    result::UpdateCheck update_result = aktualizr.CheckUpdates().get();
+    ASSERT_EQ(update_result.status, result::UpdateStatus::kError);
+  }
+}
+
+#ifndef __NO_MAIN__
+int main(int argc, char **argv) {
+  ::testing::InitGoogleTest(&argc, argv);
+
+  logger_init();
+
+  if (argc != 3) {
+    std::cerr << "Error: " << argv[0] << " requires the path to the aktualizr-repo utility "
+              << "and an OStree sysroot\n";
+    return EXIT_FAILURE;
+  }
+
+  TemporaryDirectory meta_dir;
+  TemporaryDirectory temp_sysroot;
+  sysroot = temp_sysroot / "sysroot";
+  // uses cp, as boost doesn't like to copy bad symlinks
+  int r = system((std::string("cp -r ") + argv[2] + std::string(" ") + sysroot.string()).c_str());
+  if (r != 0) {
+    return -1;
+  }
+
+  aktualizr_repo_path = argv[1];
+  Process akt_repo(aktualizr_repo_path.string());
+  akt_repo.run({"image", "--path", meta_dir.PathString(), "--targetname", "update_1.0", "--targetsha256", new_rev,
+                "--targetlength", "0", "--targetformat", "BINARY", "--hwid", "primary_hw"});
+  akt_repo.run({"addtarget", "--path", meta_dir.PathString(), "--targetname", "update_1.0", "--hwid", "primary_hw",
+                "--serial", "CA:FE:A6:D2:84:9D"});
+  akt_repo.run({"signtargets", "--path", meta_dir.PathString(), "--correlationid", "abc123"});
+  LOG_INFO << akt_repo.lastStdOut();
+  // Work around inconsistent directory naming.
+  Utils::copyDir(meta_dir.Path() / "repo/image", meta_dir.Path() / "repo/repo");
```

Comment:
I think the problem here is that this metadata is not available in the test case. I recommend putting the aktualizr-repo bits in the test case itself. The key is making sure the directory the metadata is generated in is accessible to the python server that serves it and that that server is known to the Aktualizr instance.